docs/api.md

```diff
@@ -53,13 +53,94 @@ In our experience, any system that is successful needs to grow and change as new
 
 We chose to version at the API level rather than at the resource or field level to ensure that the API presents a clear, consistent view of system resources and behavior, and to enable controlling access to end-of-lifed and/or experimental APIs.
 
-The [API and release versioning proposal](design/versioning.md) describes the current thinking on the API version evolution process.
+Note that API versioning and Software versioning are only indirectly related.  The [API and release
+versioning proposal](design/versioning.md) describes the relationship between API versioning and
+software versioning.
+
+**Please do try our beta features.  During the beta stage, we are able to make significant changes
+in interfaces based on your feedback. In later stages, changes may not be possible.** Since alpha
+features are under active development, you may wish to try alpha features in a test cluster, rather
+than in a production cluster.
+
+Different API versions imply different levels of stability and support:
+ - Alpha level:
+   - The version names contain `alpha` (e.g. `v1alpha1`).
+   - *Purpose*: sharing new features with developers and eager expert users, prior to:
+     - *complete implementation*: for instance, some API operations, CLI
+     commands, or UI support may not be implemented. 
+     - *thorough API review*: the object schema may change in a later software release, and it may be necessary
+       to delete/modify/re-create objects in order to upgrade a cluster to a subsequent software release.
+       The object modification may not be easily automated, and the process may be poorly documented.
+     - *commitment to complete the feature*: the feature may be dropped entirely in a later software release.
+     - *allow rapid changes in versions*: API versions may increment faster than the minor release cadence.
+   - Alpha features are present in released binaries, but disabled by default. Enabling an Alpha feature
+     it is be necessary to make flag changes on the API server, controller, kubelet, etc..
+     Hosted clusters (e.g. GKE) may not allow users to enable alpha object versions and features at all.
+  
+ - Beta level:
+   - The version names contain `beta` (e.g. `v2beta3`).
+   - *Purpose*: getting feedback from a wider set of users on usability and API consistency, while offering
+     enough stability to allow use in clusters running realistic loads (e.g. QA cluster for a large web service).
+   - Before graduating from alpha to beta:
+     - a feature is fully implemented; all relevant REST endpoints and kubectl commands work.  The code is tested to high standards.
+     - a feature has had an API review, though the beta period is still likely to turn up API issues not
+       found in review.
+     - the project is committed to graduating the feature to stable (typically within 3 months, but sometimes longer)
+     - has been tested to high standards and is considered safe to enable in a production cluster.
```

Comment:
i prefer this sentence to the one a few earlier - the example is too vague to derive any meaning - is a web service large in number of pods, qps, data volume, etc.  seems better to just say its safe to enable in production